### PR TITLE
Fix NPE when convert invalid results (non-exist names)

### PR DESCRIPTION
**Pull request by baev**

### CODE REVIEW

File: allure-report-data/src/test/groovy/ru/yandex/qatools/allure/data/converters/DefaultTestCaseConverterTest.groovy [hunk 1]
```diff
@@ -296,6 +299,90 @@ class DefaultTestCaseConverterTest {
...
+
+    @Test
+    void shouldConvertStepWithoutName() {
+        def origin = new TestCaseResult(
+                name: "name",
+                steps: [new Step(title: "some title")]
+        )
+        def modify = converter.convert(origin)
+
+        assert modify.steps
```
Comment (lanwen) [inline allure-report-data/src/test/groovy/ru/yandex/qatools/allure/data/converters/DefaultTestCaseConverterTest.groovy]: u can use hamcrest's assertThat() as in java. Why u don't?

Comment (baev) [inline allure-report-data/src/test/groovy/ru/yandex/qatools/allure/data/converters/DefaultTestCaseConverterTest.groovy]: Why do I need to use `assertThat`? What the reason?

Comment (lanwen) [inline allure-report-data/src/test/groovy/ru/yandex/qatools/allure/data/converters/DefaultTestCaseConverterTest.groovy]: more describable assertions?

Comment (baev) [inline allure-report-data/src/test/groovy/ru/yandex/qatools/allure/data/converters/DefaultTestCaseConverterTest.groovy]: @lanwen I think it is pretty good
```
Assertion failed: 

assert loaded.size() == countOfServicesShouldBeLoaded
       |      |      |  |
       []     0      |  1
                     false
```

Comment (lanwen) [inline allure-report-data/src/test/groovy/ru/yandex/qatools/allure/data/converters/DefaultTestCaseConverterTest.groovy]: groovy magic?

Comment (vania-pooh) [inline allure-report-data/src/test/groovy/ru/yandex/qatools/allure/data/converters/DefaultTestCaseConverterTest.groovy]: @baev I would agree with @lanwen. JUnit best practice is to use matchers. Either don't use JUnit in these tests which is not a good idea or migrate to matchers. However you could use Groovy style calls without parentheses. Something like:
```groovy
assertThat loaded hasSize 3
```

**Pull request merged**